### PR TITLE
fix: update approval logic to work

### LINKED ISSUE #34: Approval is flaky
The approve button on the craft page (the green tick) doesn't always work properly:
* Sometimes no TX is created (I think because it was clicked too soon?)
* After the TX is sent, the UI doesn't reflect the pending status immediately
* Sometimes it allows for a second TX to be sent (even though the button is disabled on pending == true)

### LINKED ISSUE #48: Add balance validation to craft
No checks are performed to test if the current balance is enough to perform a craft (the UI allows for a failing TX to be sent). Besides disabling the button, this info needs to be sent to the `CraftingIngredientRow` component so that it can render accordingly.

### PR DESCRIPTION
**Pull request by shrugs**
fixes #34 

also fixes #48

### CODE REVIEW

File: app/src/pages/CraftableTokenPage.js [hunk 1]
```diff
@@ -110,19 +108,25 @@ class CraftableTokenPage extends React.Component {
...
+      // @TODO(shrugs) - notify user of error
+      console.error(error)
+    } finally {
+      await when(() => field.$('approved').values())
       runInAction(() => {
         this.expectingChange.set(address, false)
       })
-      // cancel local pending
-      console.error(error)
+      field.$('pending').set(false)
```
Comment (nventuro) [inline app/src/pages/CraftableTokenPage.js]: What about rejections or TX failures? The `when` condition will never be true, and `pending` won't be set back to false

Comment (nventuro) [reply app/src/pages/CraftableTokenPage.js]: The whole `finally` block is a bit weird, since it only makes sense for the successful case

Comment (shrugs) [reply app/src/pages/CraftableTokenPage.js]: hmm yeah. the reason that `when` is there is to stop the UI from flashing in between `request finishes -> pending set to false, approval still false, ui updates -> mobx reloads the allowances -> ui updates to show approval granted`.

Comment (nventuro) [reply app/src/pages/CraftableTokenPage.js]: I see. I moved the `when` to the `try` block and rejections are now handled properly. We still get some flashing though, but I think it's due to how everything is updated from scratch every 15s

**Pull request merged**